src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -428,6 +441,12 @@ private static AuthorityType GetAuthorityType(string authority)
             return AuthorityType.Aad;
         }
 
+        private static bool isCiamAuthority(string authority)
```

Comment:
`IsCiamAuthority`
I think this method is more appropriate in the CiamAuthority class.

Reply:
same as this https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/4054#discussion_r1159251858

Reply:
I can take another look at this to see if there is another way to better organize this without too much refactoring. However, we are looking to release by Friday so we can prepare for MS Build so it may be a follow up PR later.

Reply:
I don't like this method because it actually does a lot of work and it can be abused. We should parse the string ONCE and only once.